spec/models/activity_spec.rb

```diff
@@ -11,18 +11,81 @@
 
   it { should validate_inclusion_of(:action).in_array %i[create update destroy] }
 
-  describe "#trackable=" do
-    context "when passed an Issue" do
-      it "sets trackable_type as Issue, not Note" do
+  describe '#trackable=' do
+    context 'when passed an Issue' do
+      it 'sets trackable_type as Issue, not Note' do
         # Default Rails behaviour is to set trackable_type to 'Note' when you
         # pass an Issue, meaning that it gets loaded as a Note, not an Issue,
         # when you call #trackable later.
         issue    = create(:issue)
         activity = create(:activity, trackable: issue)
-        expect(activity.trackable_type).to eq "Issue"
+        expect(activity.trackable_type).to eq 'Issue'
         expect(activity.reload.trackable).to eq issue
       end
     end
   end
 
+  describe 'filtering' do
+    before do
+      card = create(:card)
+      issue = create(:issue)
+      @user = create(:user)
+      activity = create(:activity, trackable: issue, user: @user, created_at: DateTime.now.beginning_of_year)
+      activity2 = create(:activity, trackable: card, user: @user, created_at: DateTime.now.beginning_of_year)
+      activity3 = create(:activity, trackable: card, created_at: DateTime.now.beginning_of_year - 1.year)
+    end
+
+    describe '#by_user' do
+      context 'when passed a valid user id' do
+        it 'returns activities of this user' do
+          expect(Activity.by_user(@user.id).count).to eq 2
+          expect(Activity.all.count).to eq 3
```

Comment:
For the sake of keeping things simple I don't think we need this check in every test, the check for the scoped records should be enough